decoder/impeghd_obj_ren_dec.c

```diff
@@ -152,13 +144,13 @@ static IA_ERRORCODE impeghd_flex_spk_2_ls_geometry(ia_cicp_ls_geo_str *array_cic
     {
       WORD32 idx = ptr_ref_spk_layout->cicp_spk_idx[i];
       pp_cicp_ls_geometry[i] = &array_cicp_ls_geometry[idx];
-      pp_cicp_ls_geometry[i]->ls_elevation = ptr_flex_spk->str_flex_spk_descr[idx].el_angle_idx;
+      pp_cicp_ls_geometry[i]->ls_elevation = ia_cicp_ls_geo_tbls[idx].ls_elevation;
       pp_cicp_ls_geometry[i]->ls_elevation_start = ia_cicp_ls_geo_tbls[idx].ls_elevation_start;
       pp_cicp_ls_geometry[i]->ls_elevation_end = ia_cicp_ls_geo_tbls[idx].ls_elevation_end;
-      pp_cicp_ls_geometry[i]->ls_azimuth = ptr_flex_spk->str_flex_spk_descr[idx].az_angle_idx;
+      pp_cicp_ls_geometry[i]->ls_azimuth = ia_cicp_ls_geo_tbls[idx].ls_azimuth;
       pp_cicp_ls_geometry[i]->ls_azimuth_start = ia_cicp_ls_geo_tbls[idx].ls_azimuth_start;
       pp_cicp_ls_geometry[i]->ls_azimuth_end = ia_cicp_ls_geo_tbls[idx].ls_azimuth_end;
-      pp_cicp_ls_geometry[i]->lfe_flag = ptr_flex_spk->str_flex_spk_descr[idx].is_lfe;
```

Comment:
If str_flex_spk_descr structure is not used, we can remove.

Reply:
str_flex_spk_descr structure has been used to populate field when spk_layout_type is 2.